File: app/src/main/java/com/future/awaker/db/dao/SpecialDetailDao.java
package com.future.awaker.db.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.future.awaker.data.SpecialDetail;

import io.reactivex.Flowable;


@Dao
public interface SpecialDetailDao {

    @Query("SELECT * FROM special_detail WHERE id = :id")
    Flowable<SpecialDetail> loadSpecialDetail(String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertSpecialDetail(SpecialDetail specialDetail);
}

File: article/src/main/java/com/ruzhan/awaker/article/db/dao/SpecialDetailDao.kt
package com.ruzhan.awaker.article.db.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query

import com.ruzhan.awaker.article.model.SpecialDetail

import io.reactivex.Flowable


@Dao
interface SpecialDetailDao {

    @Query("SELECT * FROM special_detail WHERE id = :id")
    fun loadSpecialDetail(id: String): Flowable<SpecialDetail>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertSpecialDetail(specialDetail: SpecialDetail)
}
